src/Roslyn.Diagnostics.Analyzers/CSharp/CSharpAvoidOptSuffixForNullableEnableCodeCodeFixProvider.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Rename;
+using Roslyn.Diagnostics.Analyzers;
+
+namespace Roslyn.Diagnostics.CSharp.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    [Shared]
+    public sealed class CSharpAvoidOptSuffixForNullableEnableCodeCodeFixProvider : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(CSharpAvoidOptSuffixForNullableEnableCode.Rule.Id);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var title = RoslynDiagnosticsAnalyzersResources.AvoidOptSuffixForNullableEnableCodeRuleIdTitleCodeFixTitle;
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+                if (root == null)
+                {
+                    continue;
+                }
+
+                var variable = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+                if (variable == null)
+                {
+                    continue;
+                }
+
+                var semanticModel = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+                var variableSymbol = semanticModel.GetDeclaredSymbol(variable, context.CancellationToken);
+                if (variableSymbol == null)
+                {
+                    continue;
+                }
+
+                var newName = variableSymbol.Name.Substring(0, variableSymbol.Name.Length - CSharpAvoidOptSuffixForNullableEnableCode.OptSuffix.Length);
+
+                // There is no symbol matching the new name so we can register the codefix
+                if (semanticModel.LookupSymbols(diagnostic.Location.SourceSpan.Start, variableSymbol.ContainingType, newName).IsEmpty)
```

Comment:
I am not entirely sure to understand which value should be used as `position` (first argument) of this method call.

Reply:
I think this is fine.

Reply:
Please add a test where lookup symbols will return non-empty symbols, i.e. we bail out due to name clash.

Reply:
I am having some trouble with the test. It is complaining about a mismatch of diagnostic but I have declared them at the correct locations... Any idea what's wrong?

Reply:
Also, it seems that the `SemanticModel.LookupSymbols` is only looking backward so we end-up with wrong results/state for other parameters in a method declaration. Or maybe I am not passing the right arguments to the call.

Reply:
Can you try to find the ancestor `BlockSytnax` and give the span end position as the position argument?

Reply:
I have tried with `Start` and `End` of the ancestor `BlockSyntax` and that doesn't solve the issue (on the opposite, the codefix no longer works).

Reply:
Can you file a tracking issue? I think this is possiblly a corner case enough to not block this PR.

Reply:
See #3680 3707